packages/vscode-extension/src/common/DeviceSessionsManager.ts

```diff
@@ -0,0 +1,27 @@
+import { Disposable } from "vscode";
+import { DeviceSessionDelegate } from "../project/deviceSession";
+import { DeviceInfo } from "./DeviceManager";
+
+export type DeviceSessionsManagerDelegate = DeviceSessionDelegate & {
+  onDeviceSelected: (deviceInfo: DeviceInfo) => void;
+  onReloadRequested: (type: ReloadAction) => void;
+};
+
+export type SelectDeviceOptions = {
+  preservePreviousDevice?: boolean;
+};
```

Comment:
I still dislike this.
I'd rather have the callsite call
```ts
if (!preserve) {
manager.stopDevice(oldDevice);
}
manager.selectDevice(newDevice);
```
instead of
```ts
manager.selectDevice(newDevice, { preservePreviousDevice })
```